### PR TITLE
typescriptify phenyl-mongodb

### PR DESCRIPTION
**Pull request by nicholaslee119**
typscriptify phenyl-mongodb

### CODE REVIEW

**Comment by nicholaslee119** [comment]
The `phenyl/mongodb` has dependency on `phenyl/central-state`,
so we should test `phenyl/mongodb` with  `phenyl/central-state` together after merging those into typescript branch

File: modules/mongodb/src/mongodb-client.ts [hunk 1]
```diff
@@ -94,22 +101,24 @@ export function filterFindOperation(operation: FindOperation): FindOperation {
...
-  const renameOperatorWithParent = Object.keys(renameOperator).reduce((operator, key) => {
+  const renameOperatorWithParent = Object.keys(renameOperator).reduce((operator: UpdateOperand<"$rename">, key: string) => {
     operator[key] = key.split('.').slice(0, -1).concat(renameOperator[key]).join('.')
     return operator
   }, {})
 
-  return assign(operation, { $set: { $rename: renameOperatorWithParent }})
+  const { $set, $docPath } = $bind<typeof operation>()
+  // @ts-ignore @TODO: ??
+  return update(operation, $set($docPath('$set', '$rename'), renameOperatorWithParent))
```
Comment (shinout) [inline modules/mongodb/src/mongodb-client.ts]: `set($docPath('$rename'), renameOperatorWithParent))`

File: modules/mongodb/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "repository": "phenyl-js/phenyl",
+  "license": "Apache License 2.0",
+  "author": "Shin Suzuki <shinout310@gmail.com>",
+  "files": [
+    "lib",
+    "LICENSE",
+    "README.md"
+  ],
+  "main": "lib/index.js",
+  "types": "lib/index.d.js",
```
Comment (Naturalclar) [inline modules/mongodb/package.json]: types should be changed to `lib/index.d.ts`

File: modules/mongodb/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@phenyl/mongodb",
+  "version": "1.0.0",
+  "description": "",
+  "repository": "phenyl-js/phenyl",
+  "license": "Apache License 2.0",
+  "author": "Shin Suzuki <shinout310@gmail.com>",
```
Comment (Naturalclar) [inline modules/mongodb/package.json]: add your name to author section :+1:

File: modules/mongodb/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  "scripts": {
+    "type-check": "upbin tsc --noEmit",
+    "pretest": "test/scripts/init-mongodb.sh",
+    "test": "upbin mocha --require ts-node/register 'test/**/*.ts' --color always || (npm run posttest && exit 1)",
+    "posttest": "pkill -x mongod > /dev/null; rm -rf test/tmp/mongodb"
+  },
+  "devDependencies": {
+    "@types/bson": "^4.0.0",
+    "@types/mongodb": "^3.1.20",
+    "babel-register": "^6.24.1",
```
Comment (Naturalclar) [inline modules/mongodb/package.json]: move @types devDependencies to root, and delete unnecessary devDependencies

File: modules/mongodb/src/mongodb-client.ts [hunk 5]
```diff
@@ -200,10 +212,10 @@ export class PhenylMongoDbClient<M: EntityMap> implements DbClient<M> {
         'NotFound'
       )
     }
-    return filterOutputEntity(result[0])
+    return filterOutputEntity<any>(result[0])
```
Comment (Naturalclar) [inline modules/mongodb/src/mongodb-client.ts]: add todo comment to add types

File: tsconfig.base.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
     "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
     "experimentalDecorators": true,
-    "lib": ["esnext", "dom"],
+    "lib": ["esnext", "dom", "es2017"],
```
Comment (Naturalclar) [inline tsconfig.base.json]: is this necessary? I feel like `esnext` will cover `es2017`

**Pull request merged**